tests/test_e2e.py

```diff
@@ -130,49 +107,40 @@ def test_tensorrtllm_e2e(self, protocol, setup_and_teardown):
         ],
     )
     @pytest.mark.timeout(900)
-    def test_vllm_e2e(self, protocol, setup_and_teardown):
+    def test_vllm_e2e(self, protocol):
         # NOTE: VLLM test models will be passed by the testing infrastructure.
         # Only a single model will be passed per test to enable tests to run concurrently.
         model = os.environ.get("VLLM_MODEL")
         assert model is not None, "VLLM_MODEL env var must be set!"
         self._import(model)
-        pid = utils.run_server()
-        setup_and_teardown.pid = pid
-        # vLLM will download the model on the fly, so give it a big timeout
-        # TODO: Consider one of the following
-        # (a) Pre-download and mount larger models in test environment
-        # (b) Download model from HF for vLLM at import step to remove burden
-        #     from server startup step.
-        utils.wait_for_server_ready(timeout=600)
-
-        self._infer(model, prompt=PROMPT, protocol=protocol)
-        self._profile(model, backend="vllm")
+        with utils.MockServer(timeout=600) as _:
```

Comment:
Didn't look deeply into the root cause, but something about the model repository destination seems not correct after these changes:

Creating model repo succeeded, looks like it's pointing to `./models`:
```
models/
└── gpt2/
    ├── 1/
    │   └── model.json
    └── config.pbtxt
```
but starting the server failed because it was looking for `~/models` (`/root/models`)
```
...
error: creating server: Internal - failed to stat file /root/models
```

Check out the vLLM job failures in the pipeline for this example.